torch/_inductor/fx_utils.py

```diff
@@ -87,7 +87,13 @@ def is_fake_tensor_same(new, old):
                 return all(
                     is_fake_tensor_same(new_i, old_i) for new_i, old_i in zip(new, old)
                 )
-            assert isinstance(new, torch.Tensor)
+            if not isinstance(new, torch.Tensor):
+                assert isinstance(
+                    new, (torch.SymInt, torch.SymBool, torch.SymFloat)
+                ), f"Unknown type {type(new)}"
+                return V.graph.sizevars.statically_known_equals(
```

Comment:
Looks like this is run before `V.graph` is initialized, so we'll need to call `is_expr_static_and_true` on the shape environment directly which you can get through `new.node.shape_env`.